packages/instantsearch-react/src/connectors/connectHierarchicalMenu.js

```diff
@@ -116,7 +116,7 @@ export default createConnector({
     const selectedItem = getSelectedItem(props, state);
     return {
       id,
-      filters: selectedItem === null ? [] : [{
+      filters: !selectedItem ? [] : [{
```

Comment:
Thanks for the thorough explanation :)

This is definitely tricky. Maybe best is not to slow this down, do something for consistency and keep that in mind as you are moving forward with the tests. WDYT?